### PR TITLE
Part 3 - Make base browser components disposable

### PR DESCRIPTION
**Pull request by cheenamalhotra**
More changes to follow to dispose these disposables in next PRs.

### CODE REVIEW

File: src/sql/base/browser/ui/table/asyncDataView.ts [hunk 1]
```diff
@@ -250,7 +251,8 @@ export class AsyncDataProvider<T extends Slick.SlickData> implements IDisposable
 		return this.dataRows.getLength();
 	}
 
-	dispose() {
+	override dispose() {
+		super.dispose();
 		this.dataRows.dispose();
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/table/asyncDataView.ts:256]: Could get rid of this and just register dataRows

File: src/sql/base/parts/tree/browser/treeImpl.ts [hunk 2]
```diff
@@ -302,7 +303,8 @@ export class Tree implements _.ITree {
 		return new MappedNavigator(this.model.getNavigator(fromElement, subTreeOnly), i => i && i.getElement());
 	}
 
-	public dispose(): void {
+	public override dispose(): void {
+		super.dispose();
```
Comment (Charles-Gagnon) [inline src/sql/base/parts/tree/browser/treeImpl.ts:307]: Could register all these things instead

File: src/sql/base/parts/tree/browser/treeModel.ts [hunk 3]
```diff
@@ -14,21 +14,23 @@ interface IMap<T> { [id: string]: T; }
...
 	}
 
 	get item(): Item {
 		return this._item;
 	}
 
-	dispose(): void {
+	override dispose(): void {
+		super.dispose();
 		if (this._onDispose) {
```
Comment (Charles-Gagnon) [inline src/sql/base/parts/tree/browser/treeModel.ts:34]: You're going to be disposing this before calling it. I'd suggest looking into how this is used to understand how best to dispose the emitter (it's possible what it had before is correct)

File: src/sql/base/parts/tree/browser/treeModel.ts [hunk 4]
```diff
@@ -689,7 +692,8 @@ export class Item {
 		return this._isDisposed;
 	}
 
-	public dispose(): void {
+	public override dispose(): void {
+		super.dispose();
```
Comment (Charles-Gagnon) [inline src/sql/base/parts/tree/browser/treeModel.ts:696]: The emitters are being disposed twice now

**Review by kisantia — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**